pkg/generator/generate.go

```diff
@@ -597,8 +597,9 @@ func (g *schemaGenerator) generateStructType(
 		}
 		valueType := codegen.Type(codegen.EmptyInterfaceType{})
 		var err error
-		if t.AdditionalProperties != nil {
-			if valueType, err = g.generateType(t.AdditionalProperties, nil); err != nil {
+		if b, ok := (*t.AdditionalProperties).(bool); t.AdditionalProperties != nil && (!b || ok) {
+			casted := (*t.AdditionalProperties).(*schemas.Type)
+			if valueType, err = g.generateType(casted, nil); err != nil {
```

Comment:
same goes here about splitting declaration and if statement
```suggestion
			valueType, err = g.generateType(casted, nil); 
			if err != nil {
```

Reply:
Since additionalProperties in schema can be either false or Type, then we can pass casted value and use only casting status

Reply:
how is this related to separation of declaration and if statement o.O